### DISCUSSION TITLE
Refresh the configuration generated by `biome init`

### DISCUSSION THREAD

**Discussion by Conaclos** [microsoft/FluidFramework#21199] [Ideas]
Biome currently generates the following configuration file when we execute `biome init`:

```json
{
	"organizeImports": {
		"enabled": true
	},
	"linter": {
		"enabled": true,
		"rules": {
			"recommended": true
		}
	}
}
```

I don't know the reasons for this choice. However, I think it is time to refresh it.

This generated configuration is certainly the first Biome configuration a user will see.
We should add more settings to help a user to figure out what they want to set.
Basically, we should add the more commonly used settings in the community.

I opened this discussion to discuss which settings should be included in the generated configuration.
Here is my first proposal:

```json
{
	"$schema": "https://biomejs.dev/schemas/1.8.0/schema.json",
	"files": {
		"ignoreUnknown": false,
		"ignore": []
	},
	"formatter": {
		"enabled": true,
		"identStyle": "tab"
	},
	"javascript": {
		"formatter": {
			"quoteStyle": "double"
		}
	},
	"linter": {
		"enabled": true,
		"rules": {
			"recommended": true
		}
	},
	"organizeImports": {
		"enabled": true
	},
	"vcs": {
		"enabled": false,
		"clientKind": "git",
		"useIgnoreFile": true
	}
}
```

**Comment by tylerbutler** [comment]
@Conaclos Thanks for the post! I'm glad you're looking for feedback. We are still in the process of adopting biome, and when we started either `biome init` didn't exist or we didn't know about it, so we just copy/pasted from the docs (which are quite good btw - the team's efforts there are very apparent) and updated as needed. Our current config is [here](https://github.com/microsoft/FluidFramework/blob/main/biome.json) but I have some active PRs that will update it to jsonc soon.

The defaults you suggested make sense. I would further suggest two things:

1. Consider making the default config file that `biome init` creates biome.jsonc instead of vanilla JSON...
2. ...And then add comments to the default config that's generated explaining what the settings do or linking to the appropriate docs. One particularly important thing to document inline or link to is how globs are resolved and how include/exclude work together.

Being able to comment in the config is extremely useful, and we regularly have problems keeping track of why configs are set up the way they are when we can't comment inline in the configs. In fact, I opened a PR last week on the [changesets](https://github.com/changesets/changesets) project to add support for comments-capable config formats. Using JSONC as the default format will be a better overall user experience IMO.

I have other feedback based on our adoption but it's not related to `biome init`, so I'll leave it out. If you're interested, though, I am happy to share more about our experience.
